### PR TITLE
Khadri 19092024 1216 pm

### PR DESCRIPTION
**Pull request by sgreddy123**

### CODE REVIEW

**Review by manvithagali52 — APPROVED** [review]

**Review by thasmiyas39 — APPROVED** [review]

File: src/main/java/com/khadri/jdbc/prepare/data/customer/dao/CustomerInsertDao.java [hunk 1]
```diff
@@ -2,15 +2,19 @@
...
 import java.sql.Connection;
 import java.sql.PreparedStatement;
+import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.sql.Statement;
 
 import com.khadri.jdbc.prepare.connection.JdbcConnectionUtil;
 import com.khadri.jdbc.prepare.data.customer.dao.model.Customer;
 
 public class CustomerInsertDao {
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/customer/dao/CustomerInsertDao.java]: change class name from CustomerInsertDao to CustomerDao
add the select and insert methods into it.

File: src/main/java/com/khadri/jdbc/prepare/data/customer/dao/CustomerInsertDao.java [hunk 2]
```diff
@@ -22,11 +26,31 @@ public void insertData(Customer cust) {
 			pstmt.setLong(4, cust.getPhoneNum());
 
 			pstmt.executeUpdate();
-
+
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/customer/dao/CustomerInsertDao.java:33]: what is the change here?

File: src/main/java/com/khadri/jdbc/prepare/data/customer/dao/CustomerInsertDao.java [hunk 3]
```diff
@@ -22,11 +26,31 @@ public void insertData(Customer cust) {
...
+						+ " - " + resultSet.getLong(4));
+			}
+		} catch (SQLException sqle) {
+			System.out.println(" customerSelectDao SQLException occours" + sqle);
+
 		} finally {
 			JdbcConnectionUtil.closeResources();
 		}
 	}
+
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/customer/dao/CustomerInsertDao.java:61]: what is the change here?

File: src/main/java/com/khadri/jdbc/prepare/data/customer/processor/CustomerDataProcessor.java [hunk 4]
```diff
@@ -15,21 +15,28 @@ public CustomerDataProcessor(Scanner scanner, CustomerInsertDao custDao) {
 	}
 
 	public void process(int recordNumber) {
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/customer/processor/CustomerDataProcessor.java:17]: what is the change in below lines 18 to 39 ????

File: src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java [hunk 5]
```diff
@@ -63,17 +63,37 @@ private void process() {
...
-			for (int i = 0; i < custCount; i++) {
+			for (int i = 0; i < customerCount; i++) {
 				custProcessor.process(custRowCount);
 				custRowCount++;
 			}
-		case 4:
+			System.out.println(customerCount + " records inserted successfully.");
+			} else if (operationTypeCustomer == 2) {
+			  System.out.println("fetching customer records from the database...");
+			  CustomerInsertDao custInsertDao = new CustomerInsertDao();
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java]: why are you are not using custDao which is declared and initialized in instance block  ???? instead creating an object???

File: src/main/java/com/khadri/jdbc/prepare/data/customer/dao/CustomerSelectDao.java
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/customer/dao/CustomerSelectDao.java:1]: Not required this class add the customerSelectData() into CustomerDao.java

**Pull request merged**